tensorflow/contrib/seq2seq/python/ops/attention_wrapper.py

```diff
@@ -335,6 +337,7 @@ def __init__(self,
                scale=False,
                probability_fn=None,
                score_mask_value=float("-inf"),
+               dtype=dtypes.float32,
```

Comment:
this can't be none?  can't you read it from the memory?

similarly, score_mask_value should probably be None here, documented as -inf in whatever the dtype of memory is.

in the code; do something like

memory.dtype.as_numpy_dtype(-np.inf)

Reply:
Actually, I removed dtype from BaseAttention since dtype is given in Bahdanau and Luong attentions and this dtype is used in memory and query layers. The dype is passed to the layers in the attention classes.

Default score_mask_value is None. If no score_mask_value is given, memory.dtype.as_numpy_dtype(-np.inf) is used.

If dtype given in attetions is None dtype is float32.